tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1192,4 +1192,111 @@ public function testDoesNotListIndexesImplicitlyCreatedByForeignKeys()
+     * @group 2782
+     * @group 6654
+     */
+    public function testComparatorShouldAddCommentToLegacyJsonArrayType()
+    {
+        if ( ! $this->_sm->getDatabasePlatform()->hasNativeJsonType()) {
+            $this->markTestSkipped('This test is only supported on platforms that have native JSON type.');
+        }
+
+        $this->_conn->executeQuery(
+            'CREATE TABLE json_array_test (id INT NOT NULL, parameters JSON NOT NULL, PRIMARY KEY(id))'
+        );
+
+        $table = new Table('json_array_test');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('parameters', 'json_array');
+        $table->setPrimaryKey(['id']);
+
+        $comparator = new Comparator();
+        $tableDiff  = $comparator->diffTable($this->_sm->listTableDetails('json_array_test'), $table);
+
+        $this->_sm->dropTable('json_array_test');
+
+        self::assertInstanceOf(TableDiff::class, $tableDiff);
+        self::assertSame(['comment'], $tableDiff->changedColumns['parameters']->changedProperties);
+    }
+
+    /**
+     * @group 2782
+     * @group 6654
+     */
+    public function testComparatorShouldReturnAllChangesWhenUsingLegacyJsonArrayType()
+    {
+        if ( ! $this->_sm->getDatabasePlatform()->hasNativeJsonType()) {
+            $this->markTestSkipped('This test is only supported on platforms that have native JSON type.');
+        }
+
+        $this->_conn->executeQuery(
+            'CREATE TABLE json_array_test (id INT NOT NULL, parameters JSON DEFAULT NULL, PRIMARY KEY(id))'
+        );
+
+        $table = new Table('json_array_test');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('parameters', 'json_array');
+        $table->setPrimaryKey(['id']);
+
+        $comparator = new Comparator();
+        $tableDiff  = $comparator->diffTable($this->_sm->listTableDetails('json_array_test'), $table);
+
+        $this->_sm->dropTable('json_array_test');
+
+        self::assertInstanceOf(TableDiff::class, $tableDiff);
+        self::assertSame(['notnull', 'comment'], $tableDiff->changedColumns['parameters']->changedProperties);
+    }
+
+    /**
+     * @group 2782
+     * @group 6654
+     */
+    public function testComparatorShouldNotAddCommentToJsonTypeSinceItIsTheDefaultNow()
+    {
+        if ( ! $this->_sm->getDatabasePlatform()->hasNativeJsonType()) {
+            $this->markTestSkipped('This test is only supported on platforms that have native JSON type.');
+        }
+
+        $this->_conn->executeQuery(
+            'CREATE TABLE json_test (id INT NOT NULL, parameters JSON NOT NULL, PRIMARY KEY(id))'
+        );
+
+        $table = new Table('json_test');
+        $table->addColumn('id', 'integer');
+        $table->addColumn('parameters', 'json');
+        $table->setPrimaryKey(['id']);
+
+        $comparator = new Comparator();
+        $tableDiff  = $comparator->diffTable($this->_sm->listTableDetails('json_test'), $table);
+        $this->_sm->dropTable('json_test');
```

Comment:
Should be done in `tearDown` or such: otherwise, a failure in this test can lead to other failures in colliding table names in tests